agent-inject/metrics.go

```diff
@@ -0,0 +1,74 @@
+package agent_inject
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	metricsNamespace            = "vault"
+	metricsSubsystem            = "agent_injector"
+	metricsLabelNamespace       = "namespace"
+	metricsLabelType            = "injection_type"
+	metricsLabelTypeDefault     = "default"
```

Comment:
I doubt we'll change the default anytime soon, but it might be more clear to name this label "both" or "init_and_sidecar".